.ko.yaml

```diff
@@ -1,3 +1,10 @@
 defaultBaseImage: gcr.io/distroless/static:nonroot
 baseImageOverrides:
   knative.dev/client/cmd/kn: docker.io/library/alpine:latest
+builds:
+  - id: knative.dev/client/cmd/kn
+    main: ./cmd/kn
+    ldflags:
+      - -X knative.dev/client/pkg/kn/commands/version.Version={{.Env.KN_BUILD_VERSION}}
+      - -X knative.dev/client/pkg/kn/commands/version.GitRevision={{.Env.KN_BUILD_GITREV}}
+      - -X 'knative.dev/client/pkg/kn/commands/version.BuildDate={{.Env.KN_BUILD_DATE}}'
```

Comment:
@rhuss that looks much better, even with global env vars. :) At least it's clear what we're trying to achieve here and without in-place manipulation of files.

Reply:
```console
➜  client git:(pr/fix-257) echo $KN_BUILD_VERSION $KN_BUILD_GITREV $KN_BUILD_DATE    
1.4.0 603f3f59f 2022-04-26 15:49:21
➜  client git:(pr/fix-257) podman run --rm -it ko.local/kn-0eb649368dcc185030f663a9db12e136:53980ed270a73c5b5a40e1f8d249823205e6c3705a3bea5b413691a98ff013e3 version
Version:      1.4.0
Build Date:   2022-04-26 15:49:21
Git Revision: 603f3f59f
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.31.0)
* Eventing
  - sources.knative.dev/v1 (knative-eventing v0.31.0)
  - eventing.knative.dev/v1 (knative-eventing v0.31.0)
```

Reply:
thanks, looks great !

/approve
/lgtm